src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -929,6 +932,95 @@ internal string GetPropertyRawText()
             return _parent.GetPropertyRawValueAsString(_idx);
         }
 
+        public bool ValueEquals(string text)
+        {
+            return ValueEquals(text.AsSpan());
+        }
+
+        public bool ValueEquals(ReadOnlySpan<byte> utf8Text)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+            return TextEqualsHelper(utf8Text);
+        }
+
+        public bool ValueEquals(ReadOnlySpan<char> text)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+
+            Span<byte> otherUtf8Text;
+            byte[] otherUtf8TextArray = null;
+            ReadOnlySpan<byte> utf16Text = MemoryMarshal.AsBytes(text);
+
+            int length = checked(utf16Text.Length * JsonConstants.MaxExpansionFactorWhileTranscoding);
```

Comment:
Why did this get AsBytes'd?  The length calculation is too large; that constant is for UTF-16 code units to UTF-8 code units, not UTF-16 bytes to UTF-8 bytes.

Reply:
That's a good point. This needs to be fixed in the existing `reader.TextEquals` impl as well:
https://github.com/dotnet/corefx/blob/e8990ae04e4ef62f5ccb143352472c9b4d5d5968/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L443-L445

Move the call to `AsBytes` close to the transcoding step (i.e. call to `JsonWriterHelper.ToUtf8`) and use the `otherText.Length` for the stackalloc threshold calculations.

Reply:
Please refer to this commit in the PR: https://github.com/dotnet/corefx/pull/37911/commits/c7d8d560ba9d89d39ad6413c03dfb8a2a74517ab